server/channels/store/sqlstore/team_store.go

```diff
@@ -219,7 +219,7 @@ func newSqlTeamStore(sqlStore *SqlStore) store.TeamStore {
 
 // Save adds the team to the database if a team with the same name does not already
 // exist in the database. It returns the team added if the operation is successful.
-func (s SqlTeamStore) Save(team *model.Team) (*model.Team, error) {
+func (s SqlTeamStore) Save(ctx context.Context, team *model.Team) (*model.Team, error) {
```

Comment:
Is there any reason to pass here the `context.Context` instead of the `request.CTX`?

Reply:
That was my first iteration. See [this discussion thread](https://community.mattermost.com/core/pl/11od3jpgxfgc389hnapxg4bfue).

Reply:
Yep I suggested Felipe to use `context.Context` mainly because that's what we have been doing for the other store methods. Fully agree that this is inconsistent with the app layer and needs to be fixed. However, I didn't want Felipe to be bogged down into fixing it. There needs to be a separate PR which changes all of these to `request.CTX` and also add additional boilerplate for the testhelpers and stuff.

Reply:
so... because this is the kick off of a new Open Source campaign around adding the context to the store, I would say that we need to pass the `request.CTX` here, because this is what we are aiming for, not the `context.Context` that is a pattern that we want to revert, right?